codes/csharp/chapter_divide_and_conquer/hanota.cs

```diff
@@ -0,0 +1,59 @@
+﻿/**
+* File: hanota.cs
+* Created Time: 2023-07-18
+* Author: hpstory (hpstory1024@163.com)
+*/
+
+namespace hello_algo.chapter_divide_and_conquer;
+
+public class hanota {
+    /* 移动一个圆盘 */
+    public void move(List<int> src, List<int> tar) {
+        // 从 src 顶部拿出一个圆盘
+        int pan = src[^1];
+        src.RemoveAt(src.Count - 1);
+        // 将圆盘放入 tar 顶部
+        tar.Add(pan);
+    }
+
+    /* 求解汉诺塔：问题 f(i) */
+    public void dfs(int i, List<int> src, List<int> buf, List<int> tar) {
+        // 若 src 只剩下一个圆盘，则直接将其移到 tar
+        if (i == 1) {
+            move(src, tar);
+            return;
+        }
+        // 子问题 f(i-1) ：将 src 顶部 i-1 个圆盘借助 tar 移到 buf
+        dfs(i - 1, src, tar, buf);
+        // 子问题 f(1) ：将 src 剩余一个圆盘移到 tar
+        move(src, tar);
+        // 子问题 f(i-1) ：将 buf 顶部 i-1 个圆盘借助 src 移到 tar
+        dfs(i - 1, buf, src, tar);
+    }
+
+    /* 求解汉诺塔 */
+    public void solveHanota(List<int> A, List<int> B, List<int> C) {
```

Comment:
Class member cannot have same name as the class, unless this member is a constructor in c#, so I change the method name from hanota to solveHanota.